### PR TITLE
R/add a comment 4

### LINKED ISSUE #4: Refactoring: Add a Comment
Currently, in [AddCommentPopover.jsx](https://github.com/mako-hammerhead-2022/pets-n-pats/blob/dev/client/components/AddCommentPopover.jsx#L38), the component is wired up to send the token from Auth0 along with the POST request.

However, the [apiClient function](https://github.com/mako-hammerhead-2022/pets-n-pats/blob/dev/client/apiClient/comments.js) is not wired up to receive the token and send it along to the server.

## Your job

Refactoring:
Wire up apiClient -> server route -> db function to utilize the Auth0 token and get the auth0Id and store it in the database as authorId

Testing:
Now that you've made changes to the actual code and its behaviour, adjust the corresponding tests and assert their new behaviour.

### PR DESCRIPTION
**Pull request by dhdaveidhd**
Completed the following tasks:

Refactoring:
Wire up apiClient -> server route -> db function to utilize the Auth0 token and get the auth0Id and store it in the database as authorId

Testing:
Now that you've made changes to the actual code and its behaviour, adjust the corresponding tests and assert their new behaviour.

### CODE REVIEW

File: server/routes/__tests__/comments.test.js [hunk 1]
```diff
@@ -3,26 +3,44 @@ const server = require('../../server')
...
 
 jest.mock('../../db')
+jest.mock('../../utils')
+
+beforeAll(() => {
+  jest.spyOn(console, 'log')
+  console.log.mockImplementation(() => {})
+  checkJwt.mockImplementation((req, res, next) => {
+    next()
+  })
```
Comment (joshuavial) [inline server/routes/__tests__/comments.test.js:15]: this mockImplementation needs to add something to req.user so you can test that the auth0Id was passed down to `db.insertComment` correctly

File: server/routes/__tests__/comments.test.js [hunk 2]
```diff
@@ -3,26 +3,44 @@ const server = require('../../server')
...
 
 describe('POST /api/comments', () => {
   it('adding a comment hits the db', () => {
     db.insertComment.mockReturnValue(Promise.resolve(objComment))
+
     return request(server)
       .post('/api/comments')
       .send(objComment)
       .then((res) => {
         expect(res.status).toBe(200)
```
Comment (joshuavial) [inline server/routes/__tests__/comments.test.js:30]: expecting the status to be 200 is agood start. I would also want to see some assertions that `db.insertComment` was called with the correct arguments

**Review by joshuavial — CHANGES_REQUESTED** [review]
Great work, the 'adding a comment hits the db' test just needs a few more assertions and then it's ready to go

**Comment by louis-nguyen108** [comment]
Hi JV, we have added assertion and fix the test to make sure auth0Id was passed down correctly. Can you please review?

**Review by joshuavial — APPROVED** [review]
Good to go, the req.user was being mocked in the wrong way in checkJWT

checkout https://github.com/mako-hammerhead-2022/pets-n-pats/pull/21/commits/bd5b46f92d008a0f870817e01d240d64854b84c9

**Comment by joshuavial** [comment]
The console.log also didn't need to be mocked so I removed it - https://github.com/mako-hammerhead-2022/pets-n-pats/pull/21/commits/08e12d70af3df18a4c9a99e7ffa4f819d6186cb3

**Pull request merged**